### ISSUE TITLE
FastifyDeprecation: You are decorating Request/Reply with a reference type

### ISSUE THREAD

**Issue by taosx** [fastify/fastify-request-context#30]
### 🐛 Bug Report

Using this plugin with fastify version 3.9.2 I get the the warning from the following warning:

```
[FSTDEP006] FastifyDeprecation: You are decorating Request/Reply with a reference type.
This reference is shared amongst all requests. Use onRequest hook instead. Property: requestContext
```

### ❓ Possible Fix

I found a temporary solution mentioned in another plugin:
file: **lib/requestContextPlugin.js**

```js
13:  fastify.decorateRequest('requestContext', requestContext)
```

replaced with:
```js
13:  fastify.decorateRequest('requestContext', { getter: () => requestContext })
```

### To Reproduce

- fastify v3.9.2
- fastify-request-context v2.1.2

### Your Environment

- *node version*: 15.2.1
- *fastify version*: 3.9.2
- *fastify request context version*: v2.1.2
- *os*: Linux

**Comment by kibertoad** [comment]
@taosx Thank you for reporting this, will address it shortly.

**Comment by kibertoad** [comment]
@taosx Which plugin did you see the aforemention solution in?

**Comment by kibertoad** [comment]
@taosx Thank you for the very helpful report! Fix was published in 2.1.3
